scripts/export_inspector_findings.py

```diff
@@ -169,10 +171,19 @@ def column_alpha(self, col: int) -> str:
     def findings_sort(self, item: tuple[str, list[SummaryType]]) -> tuple[int, str]:
         score = 0
         weights = {'HIGH': 1, 'CRITICAL': 10}
-        for summary in item[1]:
+        vulnerability, summaries = item
+        for summary in summaries:
             count = len(summary['resources'])
             score += count * weights.get(summary['severity'], 0)
-        return score, item[0]
+        if vulnerability.startswith('CVE-'):
+            # Best secondary-sorting effort on CVE findings, vulnerability names
+            # not prefixed with 'CVE' may reflect an inaccurate secondary order.
+            id = vulnerability.rsplit('-', 1)[1]
+            id_max_length = 6
+            if len(id) > id_max_length:
+                log.warning('Vulnerability %s may not be sorted properly', vulnerability)
+            vulnerability = vulnerability.replace(id, f'{id:0>6}')  # like …:0>id_max_length
```

Comment:
```suggestion
            # Pad with zeros the ID part of a "CVE-{YEAR}-{ID}" vulnerability
            # so we get a numeric sort of IDs when the vulnerabilities are
            # sorted alphanumerically.
            cve_year, _, id = vulnerability.rpartition('-')
            vulnerability = f'{cve_year}-{id:0>10}'
```

There is a bug with the `.replace()` approach (consider if there was a CVE `CVE-2024-202`). Also, if we just use a large enough padding (10), I think the warning isn't necessary.

Reply:
Good point, the given example isn't a bug, see https://cve.mitre.org/cve/identifiers/syntaxchange.html#new, also updated the padding to reflect the maximum CVE ID sequence length.